src/mlpack/tests/bayesian_linear_regression_test.cpp

```diff
@@ -0,0 +1,185 @@
+  BOOST_REQUIRE_SMALL(estimator.Variance(), 1e-6);
+}
+
+// Verify centerData and scaleData equal false do not affect the solution.
+BOOST_AUTO_TEST_CASE(TestCenter0ScaleData0)
+{
+  arma::mat matX;
+  arma::rowvec y;
+  size_t nDims = 30, nPoints = 100;
+
+  GenerateProblem(matX, y, nPoints, nDims, 0.5);
+
+  BayesianLinearRegression estimator(false, false);
+
+  estimator.Train(matX, y);
+
+  // Check dataOffset is empty.
+  BOOST_REQUIRE(estimator.DataOffset().n_elem == 0);
+
+  // To be neutral responseOffset must be 0.
+  BOOST_REQUIRE(estimator.ResponsesOffset() == 0);
+
+  // Check dataScale is empty.
+  BOOST_REQUIRE(estimator.DataScale().n_elem == 0);
+}
+
+// Verify that centering and normalization are correct.
+BOOST_AUTO_TEST_CASE(TestCenterDataTrueScaleDataTrue)
+{
+  arma::mat matX;
+  arma::rowvec y;
+  size_t nDims = 5, nPoints = 100;
+  GenerateProblem(matX, y, nPoints, nDims, 0.5);
+
+  BayesianLinearRegression estimator(true, true);
+  estimator.Train(matX, y);
+
+  arma::colvec xMean = arma::mean(matX, 1);
+  arma::colvec xStd = arma::stddev(matX, 0, 1);
+  double yMean = arma::mean(y);
+
+  BOOST_REQUIRE_SMALL((double) abs(sum(estimator.DataOffset() - xMean)), 1e-6);
+  BOOST_REQUIRE_SMALL((double) abs(sum(estimator.DataScale() - xStd)), 1e-6);
+  BOOST_REQUIRE_CLOSE(estimator.ResponsesOffset(), yMean, 1e-6);
+}
+
+// Make sure a model with center ans scale option set is different than a model
+// without it set.
+BOOST_AUTO_TEST_CASE(OptionsMakeModelDifferent)
+{
+  arma::mat matX;
+  arma::rowvec y;
+  size_t nDims = 10, nPoints = 100;
+  GenerateProblem(matX, y, nPoints, nDims, 0.5);
+
+  BayesianLinearRegression blr(false, false), blrC(true, false),
+      blrCS(true, true);
+
+  blr.Train(matX, y);
+  blrC.Train(matX, y);
+  blrCS.Train(matX, y);
+
+  for (size_t i = 0; i < nDims; ++i)
+    BOOST_REQUIRE((blr.Omega()(i) != blrC.Omega()(i)) &&
+                  (blr.Omega()(i) != blrCS.Omega()(i)) &&
+                  (blrC.Omega()(i) != blrCS.Omega()(i)));
+}
+
+// Check that Train() does not fail with two colinear vectors.
+BOOST_AUTO_TEST_CASE(SingularMatix)
+{
+  arma::mat matX;
+  arma::rowvec y;
+
+  GenerateProblem(matX, y, 200, 10);
+  // Now the first and the second rows are indentical.
+  matX.row(1) = matX.row(0);
+
+  BayesianLinearRegression estimator;
+  estimator.Train(matX, y);
+}
+
+// Check that std are well computed/coherent. At least higher than the
+// estimated predictive variance.
+BOOST_AUTO_TEST_CASE(PredictiveUncertainties)
+{
+  arma::mat matX;
+  arma::rowvec y;
+
+  GenerateProblem(matX, y, 100, 10, 1);
+
+  BayesianLinearRegression estimator(true, true);
+  estimator.Train(matX, y);
+
+  arma::rowvec responses, std;
+  estimator.Predict(matX, responses, std);
+  const double estStd = sqrt(estimator.Variance());
+
+  for (size_t i = 0; i < matX.n_cols; i++)
+    BOOST_REQUIRE_GT(std[i], estStd);
+
+  // Check that the estimated variance is close to 1.
+  BOOST_REQUIRE_CLOSE(estStd, 1, 20);
```

Comment:
I ran this test 1000 times with different random seeds, and found it occasionally failed (but not often), like this:

```
/home/ryan/src/mlpack/src/mlpack/tests/bayesian_linear_regression_test.cpp(158): fatal error: in "BayesianLinearRegressionTest/PredictiveUncertainties": difference{0.21546} between estStd{1.215459911581513} and 1{1} exceeds 20%
```

But the failure never exceeded 30% tolerance, so I think if you change this to 30%, it should be robust enough to not cause random failures on the build farms or for users. :+1: